### DISCUSSION TITLE
【导航菜单】部分post页的导航栏中的链接会多出一个 `[prefix]`

### DISCUSSION THREAD

**Discussion by breezedeus** [tangly1024/NotionNext#1984] [Q&A]
**描述bug**
部分post页的导航栏中的链接会多出一个 `[prefix]` 。

**复现步骤**
1. 访问：https://www.breezedeus.com/article/vie-nougat
2. 点击导航栏： Pix2Text / P2T详细资料，然后会看到链接长这样：https://www.breezedeus.com/[prefix]/pix2text_cn ，其中的 `/[prefix]` 是不应该存在的。

**期望的正常结果**
正常链接应该是：https://www.breezedeus.com/pix2text_cn

**截图**
<img width="1129" alt="image" src="https://github.com/tangly1024/NotionNext/assets/6712673/b649f609-4dbc-49c3-97e2-7ec88e6ab147">

**环境**

- 操作系统: macOS
- 浏览器 Chrome
- NotionNext版本 4.0.14
- 主题 默认

**补充说明**
感谢作者提供的开源工具，很好用👍。

**Comment by siuze** [comment]
我这边点击导航栏菜单，也会导航到多出来一个[prefix]的网址，比如应该打开：`https://notion.siuze.top/about` 却导航到`https://notion.siuze.top/[prefix]/about` 。   
这时候会加载一段时间加载不出东西，然后自动重定向到404，再自动重定向到首页刷新，此时再点击导航栏上的菜单就正常了。    
这个bug只在较长一段时间（可能缓存过期？）没访问博客后，再次访问博客时出现。按照上述过程恢复正常后，短期内不论通过什么浏览器、什么设备访问都不会有问题。所以我这会儿又恢复正常了，不好复现bug。

**Comment by tangly1024** [comment]
感谢反馈，之前某一个版本出现的bug，新版本已修复

**Reply by siuze** [reply]
可以说明一下是哪个commit修复了或者问题定位在哪里吗？我尝试合并了最新的上游代码，但是这个问题仍然存在。    

目前发现只要是在任何[文章详情页](https://notion.siuze.top/article/b5af3ca5-8f24-4e4a-b75a-4591377e4764)内，顶部导航栏的的页面链接总是会多出一个[prefix]。  

在访问首页和导航栏上的独立页面时并不会有这个问题。

**Reply by siuze** [reply]
找到原因了，似乎是旧版导航栏的to参数生成问题   
  
https://github.com/siuze/NotionNext/commit/4c7034ad226f8ee377a4b514f5efe16cf01b6707#diff-629a44078b1bac7360a3f7a0b8dce0fb9398bab58691afcae4cc5c3d42efc6b3

**Reply by erduoya77** [reply]
你好，我这好像和你得情况类似 #2420  ，但是我尝试过你的修改，还是不行
